airflow/models.py

```diff
@@ -1159,7 +1167,7 @@ def run(
         self.pool = pool or task.pool
         self.test_mode = test_mode
         self.force = force
-        self.refresh_from_db()
+        self.refresh_from_db(lock_for_update=True)
```

Comment:
Please provide the session as well, to make sure the commit happens in the same session